### ISSUE TITLE
Extract metadata from PNGs

### ISSUE THREAD

**Issue by kueda** [inaturalist/inaturalist#2706]
Apparently we're only doing it from JPGs. Not sure what the best lib to do this is, but maybe https://github.com/wvanbergen/chunky_png. Should try to extract all the same fields we get from JPGs with exifr and xmp.

**Comment by todtb** [comment]
I'm going to try to give this a go with `chunky_png`.

**Comment by todtb** [comment]
TL;DR: after some experimentation it looks like the ruby wrapper for exiftool may prove to be more useful here. I'm going to carry on using that.

---

After some experimentation it looks like `chunky_png` is using metadata stored in `zTXt`. See http://www.libpng.org/pub/png/spec/1.2/PNG-Chunks.html#C.zTXt 

This was confirmed first by converting a `.jpg` file with known exif data using `magick`(ImageMagick "preserves" exif data by shoveling into `zTXt`) and observing the fields with both [`exiftool`](https://exiftool.org/) and `chunky_png` that later of which yields:
```ruby
"exif:GPSLatitude" => "39/1, 55/1, 5801/100",
"exif:GPSLatitudeRef" => "N",
"exif:GPSLongitude" => "75/1, 12/1, 4317/100",
```
It essentially namespaces the keys in zTXt under "exif:*"

`exiftool` shows both sets of tags (zTXt top, exif bottom):

```bash
Exif GPS Latitude               : 39/1, 55/1, 5801/100
Exif GPS Latitude Ref           : N
Exif GPS Longitude              : 75/1, 12/1, 4317/100
Exif GPS Longitude Ref          : W
...
GPS Latitude                    : 39 deg 55' 58.01" N
GPS Longitude                   : 75 deg 12' 43.17" W
````

Using `exiftool` to manually add the tags to a `.png` file yields no results from `chunky`.

Luckily there is a ruby wrapper for `exiftool`: https://github.com/exiftool-rb/exiftool.rb which is showing promising results along with some useful casting of data types: 

```ruby
:gps_date_time => 2020-06-07 19:12:56 UTC,
:gps_date_time_civil => #<Date: 2020-06-07 ((2459008j,0s,0n),+0s,2299161j)>,
:gps_latitude => 39.93278056,
:gps_longitude => -75.21199167,
```

**Comment by kueda** [comment]
Cool, exiftool will work for us.

**Comment by todtb** [comment]
I've got a pretty good idea what we need to do here, but I was wondering if you had any sample images @kueda. I've been manually copying some to `.png`s which SHOULD be sufficient, however I'd like to make sure I'm testing this as close to reality as possible. A quick glance through the forums and I couldn't really find where this issue came from, was there any reference to images where this was failing?

**Comment by kueda** [comment]
This came to my attention when we decided to make it so only staff can edit the taxon photos of humans, and we wanted to ensure that metadata was visible at https://www.inaturalist.org/photos/80550183 so we could include attribution details, so here's the PNG that started that:

![Homo + Homo sapiens](https://user-images.githubusercontent.com/22691/86387192-a67aff80-bc47-11ea-9b38-120e42e73396.png)

I was mostly interested in the content of the `Description` field and any copyright fields.
